packages/compiler/core/checker.ts

```diff
@@ -107,8 +111,8 @@ export interface Checker {
   getLiteralType(node: NumericLiteralNode): NumericLiteralType;
   getLiteralType(node: BooleanLiteralNode): BooleanLiteralType;
   getLiteralType(node: LiteralNode): LiteralType;
-  getTypeName(type: Type): string;
-  getNamespaceString(type: NamespaceType | undefined): string;
+  getTypeName(type: Type, options?: TypeNameOptions): string;
```

Comment:
Not really in the scope of this PR but couldn't `getTypeName` be moved out of the checker. Doesn't feel like they depend on it.